scheduler/src/cook/scheduler/scheduler.clj

```diff
@@ -1456,10 +1458,20 @@
                                    (when-let [job (tools/job-ent->map (:job/_instance task-ent))]
                                      (let [pool-name (cached-queries/job->pool-name job)
                                            task-request (make-task-request db job pool-name :task-id task-id)
-                                           ^TaskScheduler fenzo (pool->fenzo pool-name)]
+                                           {:keys [^TaskScheduler fenzo unassign-task-set]} (pool-name->fenzo-state pool-name)
+                                           to-unassign (util/set-atom! unassign-task-set #{})
+                                           ^Action2 unassigner (.getTaskUnAssigner fenzo)]
                                        ;; Need to lock on fenzo when accessing taskAssigner because taskAssigner and
                                        ;; scheduleOnce can not be called at the same time.
                                        (locking fenzo
+                                         (doseq [[task-id hostname] to-unassign]
+                                           (try
+                                             (log/debug "In" pool-name "pool, unassigning task"
+                                                        task-id "from" hostname)
+                                             (.call unassigner task-id hostname)
+                                             (catch Exception e
+                                               (log/error e "In" pool-name "pool, failed to unassign task"
+                                                          task-id "from" hostname))))
```

Comment:
Instead of duplicating the unassignment code in `match-offer-to-schedule` and `reconcile-tasks`, can we extract a function for it?